File: TelerikUIExamples/TelerikUIExamples/ExampleInfo.m
//
//  ExampleInfo.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "ExampleInfo.h"
#import "ViewController.h"

@implementation ExampleInfo

- (instancetype)initWithExamples:(NSArray*)examples withTitle:(NSString*)title
{
    self = [self init];
    if (self) {
        _examples = examples;
        _title = title;
    }
    return self;
}

- (instancetype)initWithClass:(Class)exampleClass withTitle:(NSString*)title
{
    self = [self init];
    if (self) {
        _exampleClass = exampleClass;
        _title = title;
    }
    return self;
}

- (UIViewController *)createController
{
    UIViewController *controller = _examples ? [[ViewController alloc] initWithExample:self] : [[_exampleClass alloc] init];
    controller.title = self.title;
    return controller;
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ExampleInfo.swift
//
//  ExampleInfo.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class ExampleInfo : NSObject {
    
    let title : String
    var examples = [ExampleInfo]()
    var exampleFunc: () -> (UIViewController)
    
    init(title: String, example exampleFunc:() -> (UIViewController)) {
        self.exampleFunc = exampleFunc
        self.title = title
    }
    
    init(title: String, exampleList examples:[ExampleInfo]) {
        exampleFunc = { ViewController() }
        self.examples = examples
        self.title = title
    }
    
    func createController() -> UIViewController {
        var controller = exampleFunc()
        if examples.count > 0 {
           (controller as! ViewController).initWithExample(self)
        }
        controller.title = self.title
        return controller
    }
}
